sdk/go/pulumi/rpc.go

```diff
@@ -267,7 +267,10 @@ func marshalInputAndDetermineSecret(v interface{},
 			if rv.IsNil() {
 				return resource.PropertyValue{}, deps, secret, nil
 			}
-			v, destType = rv.Elem().Interface(), destType.Elem()
+			if destType.Kind() == reflect.Ptr {
+				destType = destType.Elem()
+			}
+			v = rv.Elem().Interface()
```

Comment:
Any test case we can/should add here?

Reply:
Happy to circle back and add one if we have a good idea on how to construct the case. I don't, so I'm going to just merge for now.